Source/WebCore/layout/formattingContexts/inline/InlineContentBalancer.cpp

```diff
@@ -110,8 +110,9 @@ InlineContentBalancer::InlineContentBalancer(InlineFormattingContext& inlineForm
 
 void InlineContentBalancer::initialize()
 {
-    if (!m_inlineFormattingContext.layoutState().placedFloats().isEmpty()) {
-        m_cannotBalanceContent = true;
+    auto numberOfVisibleLinesConsideringLineClamp = m_inlineFormattingContext.formattingUtils().numberOfVisibleLinesConsideringLineClamp();
+    if (!m_inlineFormattingContext.layoutState().placedFloats().isEmpty() || numberOfVisibleLinesConsideringLineClamp == 1) {
+        m_shouldNotBalanceContent = true;
         return;
```

Comment:
What I really meant here was that there's a difference between cases when we bail out due to unable to balance (floats are present) and cases when there's only one line and we can take a fast path. InlineContentBalancer may not care about the difference yet but I think it would make the code read better if we decoupled these 2.